### DISCUSSION TITLE
Don't download language file on each enable

### DISCUSSION THREAD

**Discussion by Andre601** [Ghost-chu/QuickShop-Reremake#1539] [Ideas]
It seems like that QuickShop as of right now is always re-downloading language files on each startup.
This has the possible chance of encountering temporary or even permanent rate limiting, depending on how Crowdin is handling those.

This could become a prominent issue if you either have a server that restarts quite a lot during the day or perhaps a network under the same domain/IP where multiple servers restart simultaneously.

One idea to solve this could be to let QuickShop have a sort of cache in which it keeps information of what it currently has vs what Crowdin has.
Downside is, that even with that would QuickShop perform another request towards crowdin, which basically keeps the risk, so perhaps the info should be retrieved from somewhere else?

I can't really think of any solution other than storing the last time a download was made and delaying any future downloads until X time has passed...

**Comment by Ghost-chu** [comment]
It should be cached once everything is up-to-date.
But there has a bug breaking the cache system.

Once it fixed, plugin should no-longer redownloading everything.

**Comment by Ghost-chu** [comment]
fixed

**Reply by Andre601** [reply]
Seems to still happen... Can't really tell tho because I don't know what the previous version of the file was.

**Reply by Ghost-chu** [reply]
> Seems to still happen... Can't really tell tho because I don't know what the previous version of the file was.

We update translation very quickly, every new commit to this repo will build a new version pf translation.

----

If you upgrade from older version, try delete cache folder to re-built cache index
